tests/plugin_http_api_test.py

```diff
@@ -791,7 +791,7 @@ def test_NetApi(self) :
         ret_json = self.nodeos.processUrllibRequest(resource, command, self.empty_content_dict, endpoint=endpoint)
         self.assertEqual(ret_json["code"], 400)
         self.assertEqual(ret_json["error"]["code"], 3200006)
-        payload = "localhost"
+        payload = "localhost:1234"
```

Comment:
Instead of modifying the test, I think you need to modify the code to continue to handle the case of no port given.